deals/module/deals.go

```diff
@@ -636,6 +650,33 @@ func notifyChanges(ctx context.Context, client *apistruct.FullNodeStruct, currSt
 	return nil
 }
 
+func robustClientGetDealInfo(ctx context.Context, lapi *apistruct.FullNodeStruct, propCid cid.Cid) (*api.DealInfo, error) {
+	di, err := lapi.ClientGetDealInfo(ctx, propCid)
+	if err != nil {
+		return nil, fmt.Errorf("client get deal info: %s", err)
+	}
+
+	// Workaround ClientGetDealInfo giving unreliable status.
```

Comment:
Kind of self-explanatory. 

If `ClientGetDealInfo` starts working correctly again, the workaround code will never get executed... so no rush to remove the workaround code. Of course, we should eventually do it to avoid dead code; but the point is that there's no rush plus no performance impact.

Reply:
Very nice.